ant-networking/src/driver.rs

```diff
@@ -1005,6 +1025,39 @@ impl SwarmDriver {
                         relay_manager.try_connecting_to_relay(&mut self.swarm, &self.bad_nodes)
                     }
                 },
+                Some(()) = Self::conditional_interval(&mut bootstrap_cache_save_interval) => {
+                    let Some(bootstrap_cache) = self.bootstrap_cache.as_mut() else {
+                        continue;
+                    };
+                    let Some(current_interval) = bootstrap_cache_save_interval.as_mut() else {
+                        continue;
+                    };
+
+                    if let Err(err) = bootstrap_cache.sync_and_save_to_disk(true).await {
```

Comment:
this will be risky as will slow down the `swarm_driver` handling, and may even result in `dead-lock`, if running multiple nodes on one machine.
`swarm_driver` shall only handle small/instant tasks. any `async` functions (potentially could be long or blocked by others) shall be passed to a spwaned task to handle, or even fired as an event to higher layer (network) to be handled.